docs/recipes/typescript.md

```diff
@@ -0,0 +1,56 @@
+# TypeScript
+
+AVA comes bundled with a TypeScript definition file. This allows developers to leverage TypeScript for writing tests.
+
+## Setup
+
+First install the TypeScript compiler [tsc](https://github.com/Microsoft/TypeScript).
+
+```
+$ npm install tsc --save-dev
+```
+
+Create a [`tsconfig.json`](https://github.com/Microsoft/TypeScript/wiki/tsconfig.json) file. This file specifies the compiler options required to compile the project or the test file.
+
+```json
+{
+	"compilerOptions": {
+		"module": "commonjs",
+		"target": "es2015"
```

Comment:
I have used `lib/test` for the TS tests and `dist/test` for the compiled tests. The options should be `"outDir": "dist", "rootDir": "lib"`.

AVA needs some configuration for that, I started AVA using `ava dist/test`. It can also be configured in the readme. Though maybe `dist/test/*.js` could be added as a new test location?